platform/android/MapboxGLAndroidSDK/src/main/java/com/mapbox/mapboxsdk/camera/CameraPosition.java

```diff
@@ -312,6 +318,10 @@ public Builder zoom(@FloatRange(from = MapboxConstants.MINIMUM_ZOOM,
      * @return CameraPosition
      */
     public CameraPosition build() {
+      if (needShift && target != null) {
+        Point point = Point.fromLngLat(target.getLongitude(), target.getLatitude());
```

Comment:
You can use `CoordinateShifterManager.getCoordinateShifter().shiftLonLat()` to make the operation less expensive. We should make it possible to check if a custom shifter is set though, to minimize the memory overhead even more - https://github.com/mapbox/mapbox-java/issues/1057.